### PR TITLE
Bun instructions for existing CRA app

### PR DESCRIPTION
**Pull request by dariocravero**
Update using bun with an existing CRA instructions in README to include a few caveats I ran into.

https://discord.com/channels/876711213126520882/887787428973281300/951179760988459048

### CODE REVIEW

**Comment by Jarred-Sumner** [comment]
Thank you for the PR!

Bun will always load  `.env.local`,  `.env.development`, and `.env` in the next release without configuration

Excluding that section, LGTM

File: README.md [hunk 1]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
 
 If no directory is specified and `./public/` doesn’t exist, bun will try `./static/`. If `./static/` does not exist, but won’t serve from a public directory. If you pass `--public-dir=./` bun will serve from the current directory, but it will check the current directory last instead of first.
 
+### Caveats of using an exsting React app:
```
Comment (timburgess) [inline README.md:237]: spelling: existing

File: README.md [hunk 2]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
...
+    "paths": {
+      "components/*": ["./src/components/*"]
+    }
+  },
+  "include": ["src"]
+}
+```
+[See this for more info on the paths key](https://code.visualstudio.com/docs/languages/jsconfig).
+
+- Add a script tag referincing `src/index.js` (or `src/index.tsx`) in `public/index.html` before the closing `</body>` tag:
```
Comment (timburgess) [inline README.md:252]: spelling: referencing

**Review by timburgess — COMMENTED** [review]
Spelling nitpicks

File: README.md [hunk 3]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
 
 If no directory is specified and `./public/` doesn’t exist, bun will try `./static/`. If `./static/` does not exist, but won’t serve from a public directory. If you pass `--public-dir=./` bun will serve from the current directory, but it will check the current directory last instead of first.
 
+### Caveats of using an exsting React app:
+- If you use [CRA's absolute imports](https://create-react-app.dev/docs/importing-a-component/#absolute-imports) add a `compilerOptions.paths` key to map them to their folders, `include` doesn't cut it, e.g.:
```
Comment (NicholasLYang) [inline README.md:238]: I'm not really clear what "`include" doesn't cut it" means. Could we expand this?

Comment (NicholasLYang) [reply README.md:238]: Also, add a comma:

```suggestion
- If you use [CRA's absolute imports](https://create-react-app.dev/docs/importing-a-component/#absolute-imports), add a `compilerOptions.paths` key to map them to their folders, `include` doesn't cut it, e.g.:
```

File: README.md [hunk 4]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
...
+}
+```
+[See this for more info on the paths key](https://code.visualstudio.com/docs/languages/jsconfig).
+
+- Add a script tag referincing `src/index.js` (or `src/index.tsx`) in `public/index.html` before the closing `</body>` tag:
+```
+    <script src="/src/index.js" async type="module"></script>
+```
+
+- Replace `%PUBLIC_URL%` references in `index.html` for nothing.
```
Comment (NicholasLYang) [inline README.md:257]: Does this mean delete these references?

File: README.md [hunk 5]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
...
+// put this in bundotenv.js and run it with node bundotenv.js
+let fs = require('fs');
+let path = require('path');
+let content = fs.readFileSync(
+  path.join(process.cwd(), '.env.development'),
+  'utf8'
+);
+let define = content
+    .split('\n')
+    .filter((item) => !item.startsWith('#'))
```
Comment (NicholasLYang) [inline README.md:270]: If a comment has leading whitespace, I don't think this works.

File: README.md [hunk 6]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
...
+    .map((item) => {
+      let [key, value] = item.split('=').map((item) => item.trim());
+      return `"process.env.${key}" = "'${value}'"`;
+    })
+    .join('\n');
+fs.writeFileSync(
+  path.join(process.cwd(), 'bunfig.toml'),
```
Comment (NicholasLYang) [inline README.md:277]: This overwrites `bunfig.toml`, which may not be what we want

File: README.md [hunk 7]
```diff
@@ -234,6 +234,54 @@ You can override the public directory by passing `--public-dir="path-to-folder"`
...
+- Add a script tag referincing `src/index.js` (or `src/index.tsx`) in `public/index.html` before the closing `</body>` tag:
+```
+    <script src="/src/index.js" async type="module"></script>
+```
+
+- Replace `%PUBLIC_URL%` references in `index.html` for nothing.
+
+- `.env` files aren't supported for CRA apps yet. A workaround is to add a `bunfig.toml` file with the variables and then run bun with `bun -c bunfig.toml` instead. Here's a script to convert `.env.development` to `bunfig.toml`:
+```
+// put this in bundotenv.js and run it with node bundotenv.js
```
Comment (NicholasLYang) [inline README.md:261]: Any reason we're not running this with bun?

**Comment by FinnRG** [comment]
@dariocravero Could you please include the requested changes? I think this could be a good improvement to the docs.

**Comment by Electroid** [comment]
Thanks for working on this! Closing this out as an old pull request, since the README has changed and we have [documentation](https://bun.sh/docs) now.

**Pull request closed without merging**